spec/models/reservation_spec.rb

```diff
@@ -87,705 +73,830 @@
+        model = EquipmentModel.find(procedure.equipment_model.id)
+        item = FactoryGirl.build_stubbed(:equipment_item,
+                                         equipment_model: model)
+        res = FactoryGirl.build_stubbed(:valid_reservation,
+                                        equipment_model: model)
+        procedures = { procedure.id.to_s => '1' }
+        allow(res).to receive(:make_notes).with('Checked out', '', [], handler)
+        res.checkout(item, handler, procedures, '')
+        expect(res).to have_received(:make_notes)
+          .with('Checked out', '', [], handler)
+      end
     end
-
-    it 'returns 0 if not overdue' do
-      reservation.update_attributes(due_date: Time.zone.today + 1.day,
-                                    overdue: false)
-      expect(reservation.late_fee).to eq(0)
+    context 'with incomplete procedures' do
+      it 'sends the incomplete procedures to the notes' do
+        # FIXME: this part of the method is really in need of refactoring
+        procedure = FactoryGirl.create(:checkout_procedure)
+        model = EquipmentModel.find(procedure.equipment_model.id)
+        item = FactoryGirl.build_stubbed(:equipment_item,
+                                         equipment_model: model)
+        res = FactoryGirl.build_stubbed(:valid_reservation,
+                                        equipment_model: model)
+        incomplete = [procedure.step]
+        allow(res).to receive(:make_notes)
+          .with('Checked out', '', incomplete, handler)
+        res.checkout(item, handler, {}, '')
+        expect(res).to have_received(:make_notes)
+          .with('Checked out', '', incomplete, handler)
+      end
     end
+  end
 
-    it 'returns the cap if a cap is set' do
-      reservation.equipment_model.update_attributes(late_fee: 100,
-                                                    late_fee_max: 100)
-      reservation.update_attributes(due_date: Time.zone.today - 3.days)
-      expect(reservation.late_fee).to \
-        eq(reservation.equipment_model.late_fee_max)
+  describe '#archive' do
+    let!(:archiver) { FactoryGirl.build_stubbed(:user) }
+    it 'returns the reservation' do
+      res = FactoryGirl.build_stubbed(:valid_reservation)
+      expect(res.archive(archiver, '')).to eq(res)
+    end
+    context 'not checked in' do
+      context 'not checked out' do
+        it 'sets checked_out to now' do
+          travel(-1.days) do
+            res = FactoryGirl.build_stubbed(:valid_reservation)
+            expect { res.archive(archiver, '') }.to change { res.checked_out }
+              .from(nil).to(Time.zone.now)
+          end
+        end
+      end
+      it 'sets checked_in to now' do
+        travel(-1.days) do
+          res = FactoryGirl.build_stubbed(:checked_out_reservation)
+          expect { res.archive(archiver, '') }.to change { res.checked_in }
+            .from(nil).to(Time.zone.now)
+        end
+      end
+      it 'updates the notes' do
```

Comment:
Would we want to test this the same way as we were testing the other two (i.e. check that we send `res` the message `make_notes`?